src/stories/common/UserNavigationMenu.stories.tsx

```diff
@@ -45,31 +81,48 @@ const UserNavigationMenuNotConnected = () => {
   );
 
   const isWalletConnected = false;
-  const [isButtonVisible, setIsButtonVisible] = useState(true);
 
   return (
     <Router>
       <div className="w-full flex justify-end relative">
         <div className="flex items-center gap-1">
           {isButtonVisible && (
-            <Button mode="tertiaryOutline" isFullRounded>
-              <Icon name="cardholder" appearance={{ size: 'tiny' }} />
-              <p className="text-sm font-inter font-medium ml-1">{formatMessage({ id: 'connectWallet' })}</p>
-            </Button>
-          )}
-          <div>
             <Button
-              className={clsx({
-                'px-4 py-2.5 !border-base-white': visible && isMobile,
-                'p-0': !visible && isMobile,
-              })}
               mode="tertiaryOutline"
               isFullRounded
-              setTriggerRef={setTriggerRef}
-              onClick={() => isMobile && setIsButtonVisible((prevState) => !prevState)}
+              setTriggerRef={setWalletTriggerRef}
+              onClick={() => isMobile && setIsWalletButtonVisible((prevState) => !prevState)}
+              className={clsx({
+                'px-4 py-2.5 !border-base-white': walletVisible && isMobile,
+                'p-0': !walletVisible && isMobile,
+              })}
             >
-              <Icon name={visible && isMobile ? 'close' : 'list'} appearance={{ size: 'tiny' }} />
+              <Icon name={walletVisible && isMobile ? 'close' : 'cardholder'} appearance={{ size: 'tiny' }} />
+              {isWalletButtonVisible && (
+                <p className="text-sm font-inter font-medium ml-1">{formatMessage({ id: 'connectWallet' })}</p>
```

Comment:
I am not sure if we need font-inter here. It is set in global styles.

Reply:
Removed